src/scripts/main.js

```diff
@@ -1,7 +1,212 @@
 'use strict';
+/* eslint-disable */
+const refs = {
+  startBtn: document.querySelector('.start'),
+  fieldCells: document.querySelectorAll('.field-cell'),
+  fieldRows: document.querySelectorAll('.field-row'),
+  messageStart: document.querySelector('.message-start'),
+  messageLose: document.querySelector('.message-lose'),
+  messageWin: document.querySelector('.message-win'),
+  score: document.querySelector('.game-score'),
+};
 
-// Uncomment the next lines to use your game instance in the browser
-// const Game = require('../modules/Game.class');
-// const game = new Game();
+const WIN_VALUE = 2048;
+const FIELD_LENGTH = 15;
+const NEW_CELL_VALUE = 2;
+const RARE_NEW_CELL_VALUE = 4;
+const RARE_CELL_CHANCE = 10;
 
-// Write your code here
+const inputHandler = ({ key }) => {
+  moveCells(key);
+};
+
+refs.startBtn.addEventListener('click', newGame);
+
+function newGame() {
+  resetField();
+  addCell(getRandomCell(), getToken());
+  addCell(getRandomCell(), getToken());
+  window.addEventListener('keydown', inputHandler);
+}
+
+function getRandomCell() {
+  if (!hasFreeSpace()) {
+    return -1;
+  }
+
+  let randomCell = getRandomNum(FIELD_LENGTH);
+
+  while (refs.fieldCells[randomCell].textContent !== '') {
```

Comment:
Using a while loop to find an empty cell can be inefficient if the field is almost full. Consider implementing a more efficient method to find empty cells.